### PR TITLE
Super Basic Initial Commenting Support 

### PR DESCRIPTION
**Pull request by Sn0wCrack**

### CODE REVIEW

File: nyaa/routes.py [hunk 1]
```diff
@@ -48,9 +48,10 @@ def modify_query(**new_values):
 
     return '{}?{}'.format(flask.request.path, url_encode(args))
 
+
 @app.template_global()
 def filter_truthy(input_list):
-    ''' Jinja2 can't into list comprehension so this is for
+    ''' Jinja2 can't into list comprehension so this igit cs for
```
Comment (nyaadev) [inline nyaa/routes.py]: what?

File: nyaa/routes.py [hunk 2]
```diff
@@ -470,16 +471,30 @@ def upload():
         return flask.render_template('upload.html', form=form, user=flask.g.user), status_code
 
 
-@app.route('/view/<int:torrent_id>')
+@app.route('/view/<int:torrent_id>', methods=['GET', 'POST'])
 def view_torrent(torrent_id):
+    current_user = flask.g.user
```
Comment (nyaadev) [inline nyaa/routes.py]: No check if the user is even logged in?

File: nyaa/routes.py [hunk 3]
```diff
@@ -470,16 +471,30 @@ def upload():
...
+    if flask.request.method == 'POST' and form.validate():
+        comment_text = (form.comment.data or '').strip()
+
+        comment = models.Comment(
+            torrent=torrent_id,
+            user_id=current_user.id,
+            text=comment_text)
+
+        db.session.add(comment)
+        db.session.flush()
```
Comment (nyaadev) [inline nyaa/routes.py]: flush() is not needed here.
Used in other places to get the id of the new row.

File: nyaa/models.py [hunk 4]
```diff
@@ -271,6 +276,23 @@ def by_category_ids(cls, main_cat_id, sub_cat_id):
...
+class Comment(db.Model):
+    __tablename__ = DB_TABLE_PREFIX + 'comments'
+
+    id = db.Column(db.Integer, primary_key=True)
+    torrent = db.Column(db.Integer, db.ForeignKey(
+        DB_TABLE_PREFIX + 'torrents.id'))
+    user_id = db.Column(db.Integer, db.ForeignKey(
+        'users.id'))
+    text = db.Column(DescriptionTextType(collation=COL_UTF8MB4_BIN), nullable=False)
+    deleted = db.Column(db.Boolean, nullable=False, default=False)
```
Comment (nyaadev) [inline nyaa/models.py]: Hard delete would be better.
Since we'd really only be deleting spam.

File: nyaa/models.py [hunk 5]
```diff
@@ -271,6 +276,23 @@ def by_category_ids(cls, main_cat_id, sub_cat_id):
         return cls.query.filter(cls.id == sub_cat_id, cls.main_category_id == main_cat_id).first()
 
 
+class Comment(db.Model):
+    __tablename__ = DB_TABLE_PREFIX + 'comments'
+
+    id = db.Column(db.Integer, primary_key=True)
```
Comment (nyaadev) [inline nyaa/models.py]: Maybe make the comment id a composite key of torrent_id + comment_id ?
So comments in a torrent start from 1

File: nyaa/models.py [hunk 6]
```diff
@@ -271,6 +276,23 @@ def by_category_ids(cls, main_cat_id, sub_cat_id):
...
+    torrent = db.Column(db.Integer, db.ForeignKey(
+        DB_TABLE_PREFIX + 'torrents.id'))
+    user_id = db.Column(db.Integer, db.ForeignKey(
```
Comment (nyaadev) [inline nyaa/models.py]: maybe add ondelete='cascade' on the foreignkey so deleting a user deletes all of his comments too.

File: nyaa/forms.py [hunk 7]
```diff
@@ -123,6 +123,13 @@ def pre_validate(self, form):
             raise ValueError(self.gettext('Not a valid choice'))
 
 
+class CommentForm(FlaskForm):
+    comment = TextAreaField('Make a comment', [
+        Length(max=10 * 1024, message='Comment must be at most %(max)d characters long.'),
```
Comment (nyaadev) [inline nyaa/forms.py]: Do we really want comments this big?
I think 255 is plenty.

It's not like people do anime reviews in the comment section, or do they?

File: nyaa/api_handler.py [hunk 8]
```diff
@@ -297,14 +297,15 @@ def api_upload(upload_request):
             # Store tracker refs in DB
             for order, tracker in enumerate(db_trackers):
                 torrent_tracker = models.TorrentTrackers(torrent_id=torrent.id,
-                    tracker_id=tracker.id, order=order)
+                                                         tracker_id=tracker.id, order=order)
```
Comment (TheAMM) [inline nyaa/api_handler.py]: There are still irrelevant changes in this commit!

File: nyaa/routes.py [hunk 9]
```diff
@@ -470,16 +471,30 @@ def upload():
         return flask.render_template('upload.html', form=form, user=flask.g.user), status_code
 
 
-@app.route('/view/<int:torrent_id>')
+@app.route('/view/<int:torrent_id>', methods=['GET', 'POST'])
```
Comment (TheAMM) [inline nyaa/routes.py]: The POST route for the comments should be different; just something like `/view/<int:torrent_id>/submit_comment` etc - let's not hijack the main view.

**Pull request merged**